src/bench.h

```diff
@@ -125,6 +125,32 @@ int have_flag(int argc, char** argv, char *flag) {
     return 0;
 }
 
+/* takes an array containing the arguments that the user is allowed to enter on the command-line
+   returns:
+      - 1 if the user entered an invalid argument
+      - 0 if all the user entered arguments are valid */
+int have_invalid_args(int argc, char** argv, char** valid_args, size_t n) {
+    size_t i;
+    int found_valid;
+    char** argm = argv + argc;
+    argv++;
+
+    while (argv != NULL && argv != argm) {
```

Comment:
`argv` will never be NULL, as it is the result of applying `argv++;` one or more times on the input. Did you mean to write `*argv != NULL` here?

Reply:
Nice catch! 
I used `argv != NULL` since the [`have_flag`](https://github.com/bitcoin-core/secp256k1/blob/master/src/bench.h#L116) function used the same loop condition to traverse user arguments.

`*argv != NULL` should be used (as you mentioned) in both this function and the `have_flag` function.

Reply:
I initially assumed `argv` will behave as a normal 2d array so, I thought both `argv` and `*argv` will value will be the same (i.e. base address of argument string). That does not seem to be the case.

Reply:
The type of `argv` is `char**`, so it's a pointer to pointer, not an array.